src/Sulu/Bundle/MediaBundle/Media/Storage/AbstractStorage.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MediaBundle\Media\Storage;
+
+/**
+ * Abstract class with basic Storage functions.
+ */
+abstract class AbstractStorage implements StorageInterface
+{
+    /**
+     * @var \stdClass
+     */
+    protected $storageOptions;
+
+    /**
+     * Add a value to the current storageOptions.
+     *
+     * @param $key
+     * @param $value
+     */
+    protected function addStorageOption($key, $value)
+    {
+        $this->storageOptions->$key = $value;
+    }
+
+    /**
+     * Returns the a value from the current storageOption.
+     *
+     * @param $key
+     *
+     * @return mixed
+     */
+    protected function getStorageOption($key)
+    {
+        return isset($this->storageOptions->$key) ? $this->storageOptions->$key : null;
+    }
+
+    /**
+     * get a unique filename in path.
+     *
+     * @param $folder
+     * @param $fileName
+     * @param int $counter
+     *
+     * @return string
+     */
+    protected function getUniqueFileName($folder, $fileName, $counter = 0)
+    {
+        $newFileName = $fileName;
+
+        if ($counter > 0) {
+            $fileNameParts = explode('.', $fileName, 2);
```

Comment:
just an example which a developer know.
did have project where i think it was CAD files or something which the client need this pre extension file types else the file could not be opened in their application